### ISSUE TITLE
Weird autocorrect with Layout/IndentFirstArrayElement when using tabs

### ISSUE THREAD

**Issue by herwinw** [rubocop-hq/rubocop#7263]
I have a legacy app that uses tabs instead of spaces, and the upgrade to Rubocop 0.74.0 started to give false positives for `Layout/IndentFirstArrayElement`.
I've created a repository at https://github.com/herwinw/rubocop_layout_indentfirstarrayelement with a minimal reproduction case.

--------

## Expected behavior

An example of code that breaks:
```ruby
def self.foobar
  [
    1,
    2,
    3
  ].each do |x|
    p x
  end
end
```
This code is aligned with tabs. The supporting config:
```yaml
AllCops:
  TargetRubyVersion: 2.3

Layout/Tab:
  Enabled: false

Layout/IndentationWidth:
  Width: 1
```

According to Rubocop 0.73.0, this is fine, but 0.74.0 started throwing errors:
```
foo.rb:11:1: C: Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
		].each do |x|
^^
```

Running autocorrect with only `Layout/SpaceInsideArrayLiteralBrackets` removes all whitespace at the beginning on this line. Running autocorrect with all cops actually runs two cops:
```
foo.rb:11:1: C: [Corrected] Layout/IndentFirstArrayElement: Indent the right bracket the same as the start of the line where the left bracket is.
].each do |x|
^
foo.rb:11:1: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
		].each do |x|
^^
```

But now my two starting tabs have been replaced by two spaces, which makes the code very inconsistent.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.

```
$ bundle exec rubocop -V
0.74.0 (using Parser 2.6.3.0, running on ruby 2.3.3 i386-linux-gnu)
```

**Comment by buehmann** [comment]
Thanks for the report. At first glance this looks like an old bug that  got unveiled by a recent change I made. Will take a closer look soon.

**Comment by herwinw** [comment]
Just to be complete: I've deleted the repository with the reproduction case, since that one is no longer needed
